src/multirust-cli/common.rs

```diff
@@ -0,0 +1,154 @@
+//! Just a dumping ground for cli stuff
+
+use multirust::{Cfg, Result, Notification, Toolchain, Error};
+use multirust_utils::{self, utils};
+use multirust_utils::notify::NotificationLevel;
+use std::ffi::OsStr;
+use std::io::{self, Write, Read, BufRead};
+use std::process::{self, Command};
+use std;
+use tty;
+use term;
+
+pub fn confirm(question: &str, default: bool) -> Result<bool> {
+    print!("{} ", question);
+    let _ = std::io::stdout().flush();
+    let input = try!(read_line());
+
+    let r = match &*input {
+        "y" | "Y" => true,
+        "n" | "N" => false,
+        "" => default,
+        _ => false,
+    };
+
+    println!("");
+
+    Ok(r)
+}
+
+fn read_line() -> Result<String> {
+    let stdin = std::io::stdin();
+    let stdin = stdin.lock();
+    let mut lines = stdin.lines();
+    lines.next().and_then(|l| l.ok()).ok_or(Error::ReadStdin)
+}
+
+pub fn wait_for_keypress() -> Result<()> {
+    let stdin = std::io::stdin();
+    if stdin.bytes().next().is_some() {
+        Ok(())
+    } else {
+        Err(Error::ReadStdin)
+    }
+}
+
+pub fn set_globals(verbose: bool) -> Result<Cfg> {
+    use download_tracker::DownloadTracker;
+    use std::cell::RefCell;
+
+    let download_tracker = RefCell::new(DownloadTracker::new());
+
+    Cfg::from_env(shared_ntfy!(move |n: Notification| {
+        if download_tracker.borrow_mut().handle_notification(&n) {
+            return;
+        }
+
+        match n.level() {
+            NotificationLevel::Verbose => {
+                if verbose {
+                    verbose!("{}", n);
+                }
+            }
+            NotificationLevel::Info => {
+                info!("{}", n);
+            }
+            NotificationLevel::Warn => {
+                warn!("{}", n);
+            }
+            NotificationLevel::Error => {
+                err!("{}", n);
+            }
+        }
+    }))
+
+}
+
+pub fn run_inner<S: AsRef<OsStr>>(mut command: Command,
+                                  args: &[S]) -> Result<()> {
+    command.args(&args[1..]);
+    // FIXME rust-lang/rust#32254. It's not clear to me
+    // when and why this is needed.
+    command.stdin(process::Stdio::inherit());
+    match exec(command) {
+        Ok(code) => {
+            // Ensure correct exit code is returned
+            process::exit(code);
+        }
+        Err(e) => {
+            Err(multirust_utils::Error::RunningCommand {
+                name: args[0].as_ref().to_owned(),
+                error: multirust_utils::raw::CommandError::Io(e),
+            }.into())
+        }
+    }
+}
+
+#[cfg(unix)]
+fn exec(mut command: Command) -> io::Result<i32> {
+    use std::os::unix::process::CommandExt;
+    Err(command.exec())
```

Comment:
from our discussion tonight, if you wanna switch this back to *not* exec then I'd also be fine with that.